internal/pkg/cli/svc_deploy.go

```diff
@@ -286,38 +290,49 @@ func (o *deploySvcOpts) configureClients() error {
 }
 
 func (o *deploySvcOpts) pushToECRRepo() error {
-	path, err := o.getDockerfilePath()
+
+	dockerBuildInput, err := o.getBuildArgs()
 	if err != nil {
 		return err
 	}
 
-	if err := o.imageBuilderPusher.BuildAndPush(docker.New(), path, o.ImageTag); err != nil {
+	if err := o.imageBuilderPusher.BuildAndPush(docker.New(), dockerBuildInput); err != nil {
 		return fmt.Errorf("build and push image: %w", err)
 	}
 
 	return nil
 }
 
-func (o *deploySvcOpts) getDockerfilePath() (string, error) {
-	type dfPath interface {
-		DockerfilePath() string
+func (o *deploySvcOpts) getBuildArgs() (*docker.BuildArguments, error) {
+	type dfArgs interface {
+		BuildArgs(rootDirectory string) *manifest.DockerBuildArgs
 	}
 
 	manifestBytes, err := o.ws.ReadServiceManifest(o.Name)
 	if err != nil {
-		return "", fmt.Errorf("read manifest file %s: %w", o.Name, err)
+		return nil, fmt.Errorf("read manifest file %s: %w", o.Name, err)
 	}
-
-	svc, err := manifest.UnmarshalService(manifestBytes)
+	svc, err := o.unmarshal(manifestBytes)
 	if err != nil {
-		return "", fmt.Errorf("unmarshal svc manifest: %w", err)
+		return nil, fmt.Errorf("unmarshal service %s manifest: %w", o.Name, err)
 	}
-
-	mf, ok := svc.(dfPath)
+	mf, ok := svc.(dfArgs)
 	if !ok {
-		return "", fmt.Errorf("service %s does not have a dockerfile path", o.Name)
+		return nil, fmt.Errorf("service %s does not have required method Build()", o.Name)
```

Comment:
This is a weird error because it specifically has to do with the interface conversion to the dfArgs interface and making sure that the unmarshaled yaml fits into a` BackendService` or `LoadBalancedWebService` struct with the right methods attached.